src/gui/sdl_mapper.cpp

```diff
@@ -2130,99 +2138,407 @@ static CKeyEvent * caps_lock_event=NULL;
+	AddKeyButtonEvent(pos_x(XO + 1),
+	                  pos_y(YO + 2),
+	                  button_width,
+	                  button_height,
+	                  "END",
+	                  "end",
+	                  KBD_end);
+	AddKeyButtonEvent(pos_x(XO + 2),
+	                  pos_y(YO + 2),
+	                  button_width,
+	                  button_height,
+	                  "PDN",
+	                  "pagedown",
+	                  KBD_pagedown);
+	AddKeyButtonEvent(pos_x(XO + 1),
+	                  pos_y(YO + 4),
+	                  button_width,
+	                  button_height,
+	                  "\x18",
+	                  "up",
+	                  KBD_up);
+	AddKeyButtonEvent(pos_x(XO + 0),
+	                  pos_y(YO + 5),
+	                  button_width,
+	                  button_height,
+	                  "\x1B",
+	                  "left",
+	                  KBD_left);
+	AddKeyButtonEvent(pos_x(XO + 1),
+	                  pos_y(YO + 5),
+	                  button_width,
+	                  button_height,
+	                  "\x19",
+	                  "down",
+	                  KBD_down);
+	AddKeyButtonEvent(pos_x(XO + 2),
+	                  pos_y(YO + 5),
+	                  button_width,
+	                  button_height,
+	                  "\x1A",
+	                  "right",
+	                  KBD_right);
 #undef XO
 #undef YO
 #define XO 0
 #define YO 7
 	/* Numeric KeyPad */
-	num_lock_event=AddKeyButtonEvent(PX(XO),PY(YO),BW,BH,"NUM","numlock",KBD_numlock);
-	AddKeyButtonEvent(PX(XO+1),PY(YO),BW,BH,"/","kp_divide",KBD_kpdivide);
-	AddKeyButtonEvent(PX(XO+2),PY(YO),BW,BH,"*","kp_multiply",KBD_kpmultiply);
-	AddKeyButtonEvent(PX(XO+3),PY(YO),BW,BH,"-","kp_minus",KBD_kpminus);
-	AddKeyButtonEvent(PX(XO+0),PY(YO+1),BW,BH,"7","kp_7",KBD_kp7);
-	AddKeyButtonEvent(PX(XO+1),PY(YO+1),BW,BH,"8","kp_8",KBD_kp8);
-	AddKeyButtonEvent(PX(XO+2),PY(YO+1),BW,BH,"9","kp_9",KBD_kp9);
-	AddKeyButtonEvent(PX(XO+3),PY(YO+1),BW,BH*2,"+","kp_plus",KBD_kpplus);
-	AddKeyButtonEvent(PX(XO),PY(YO+2),BW,BH,"4","kp_4",KBD_kp4);
-	AddKeyButtonEvent(PX(XO+1),PY(YO+2),BW,BH,"5","kp_5",KBD_kp5);
-	AddKeyButtonEvent(PX(XO+2),PY(YO+2),BW,BH,"6","kp_6",KBD_kp6);
-	AddKeyButtonEvent(PX(XO+0),PY(YO+3),BW,BH,"1","kp_1",KBD_kp1);
-	AddKeyButtonEvent(PX(XO+1),PY(YO+3),BW,BH,"2","kp_2",KBD_kp2);
-	AddKeyButtonEvent(PX(XO+2),PY(YO+3),BW,BH,"3","kp_3",KBD_kp3);
-	AddKeyButtonEvent(PX(XO+3),PY(YO+3),BW,BH*2,"ENT","kp_enter",KBD_kpenter);
-	AddKeyButtonEvent(PX(XO),PY(YO+4),BW*2,BH,"0","kp_0",KBD_kp0);
-	AddKeyButtonEvent(PX(XO+2),PY(YO+4),BW,BH,".","kp_period",KBD_kpperiod);
+	num_lock_event = AddKeyButtonEvent(pos_x(XO),
+	                                   pos_y(YO),
+	                                   button_width,
+	                                   button_height,
+	                                   "NUM",
+	                                   "numlock",
+	                                   KBD_numlock);
+	AddKeyButtonEvent(pos_x(XO + 1),
```

Comment:
IMHO each such a call should occupy a single line, even if this is against the coding rules. I suggest to format the code manually (align arguments one after another) and disable CLang checks for this piece of code.

Reply:
Hmm, I do not agree with disregarding code formatting rules, but I do agree with having a single line per call. To satisfy both, I suggest rewriting this code - first create and initialize `std::vector` or `std::array` with all parameters (each row containing a set of parameters), then iterate over that array and call `AddKeyButtonEvent` for each row.

Reply:
@dreamer In my opinion table formatted as here: https://github.com/dosbox-staging/dosbox-staging/blob/0af69f915131a0344abc1e6c9d8f105b466ce751/src/dos/dos.cpp#L85 is much more readable and easier to maintain if it does not follow certain coding rules (like max 80 characters in a line, or the usual space/comma formatting).

Reply:
Yup, column-by-column alignment is more readable when there are many rows of the same value types (and quantities of values, horizontally), like in both cases you mentioned @FeralChild64 -- especially when quantities and formatting would wrap and get very cluttered.

Perhaps with dreamer's suggestion, the code would fit on one like and be quite readable too. 

Ideally clang would have a  smart-grid-detection, that would know when it happens upon huge "grids of like values", and use column alignment, but currently it doesn't.

Reply:
If code formatted in a table does not format nicely then we have an option of selectively turning clang-format off, see: https://github.com/dosbox-staging/dosbox-staging/blob/main/src/dos/dos_keyboard_layout.cpp#L1228

But please, do not abuse this for actual code.

Reply:
I agree with @dreamer's view: we almost never should use the override for code; that should be basically almost exclusively reserved for tabular lists of values. I find the clang formatter valuable actually in these scenarios: it highlights when your lines are too high because of nesting or other reasons.

Another related issue: I really would not like to see functions with more than 3-4 paramters in new code. Arguably, if you go above that, you're much better served by simply passing in a struct.

Reply:
Was left in single-line format, see #2763. If anyone wants to change that, be my guest. But it's completely out of scope for the work done here, plus the whole mapper will be eventually rewritten anyway.